### PR TITLE
Home year selector #177

### LINKED ISSUE #177: Add Month/Year select(s) to Home Profile Activities section
As a manager, I would like to be able to answer the following questions:

- Who has been active with our residents in a given Month and Group Home?
- What are the counts of activities by type in a given Month and Group Home?

# Task
Add a month/year selection to the Home Profile in the Activities section. When a month/year combination is selected, update the Home Activities charts with data from that period. The query should default to the current month, rather than the 'last 30 days'.

### PR DESCRIPTION
**Pull request by ErikPel**
Closes #177

### CODE REVIEW

**Comment by brylie** [comment]
Hei @ErikPel, is this ready for review?

**Comment by brylie** [comment]
@ErikPel still waiting for confirmation that this is ready for review. Please resolve conflicts and mark as 'ready for review'.

**Comment by ErikPel** [comment]
Oh sorry I must have missed the email notification. Should be fine now

File: app/client/views/home/activities/counts/facilitator_role/facilitator_role.js [hunk 1]
```diff
@@ -3,64 +3,83 @@ Template.homeActivityCountsByFacilitatorRole.onCreated(function () {
...
 
-Template.homeActivityCountsByFacilitatorRole.onCreated(function () {
-  // Get reference to template instance
-  const templateInstance = this;
-
+Template.homeActivityCountsByFacilitatorRole.onRendered(function () {
+    // Get reference to template instance
+    const templateInstance = this;
+    const svg = dimple.newSvg("#homeActivityCountsByFacilitatorRoleChart", "100%", "100%");
+    console.log(svg)
```
Comment (brylie) [inline app/client/views/home/activities/counts/facilitator_role/facilitator_role.js]: Clean up console.log.

File: app/client/views/home/activities/sums/sums.js [hunk 2]
```diff
@@ -1,59 +1,79 @@
 Template.homeResidentActivitySumsByType.created = function () {
-  // Get reference to template instance
-  var instance = this;
+    // Get reference to template instance
+    var instance = this;
+    // Get current home ID
+    instance.homeId = Router.current().params.homeId;
+    // set up home resident activity sums by type reactive variable
+    instance.homeResidentsActivitySumsByType = new ReactiveVar();
+    //instance.date = instance.data.date;
```
Comment (brylie) [inline app/client/views/home/activities/sums/sums.js]: Clean up commented code

File: app/client/views/home/activities/sums/sums.js [hunk 3]
```diff
@@ -1,59 +1,79 @@
...
+    // Get home resident activity sums from server method
+    Meteor.call('getHomeResidentsActivitySumsByType', instance.homeId,instance.data.date.get(), function (error, activitySums) {
```
Comment (brylie) [inline app/client/views/home/activities/sums/sums.js]: This duplicate call to `getHomeResidentsActivitySumsByType` shouldn't be necessary, as `autorun` should run once in init.

File: app/client/views/home/activities/sums/sums.js [hunk 4]
```diff
@@ -1,59 +1,79 @@
...
+        .color(customColors)
+
+    activityCountChart
+        .yAxis.axisLabel("Activity count")
+        .tickFormat(d3.format('d'));
+
+    instance.autorun(function () {
+        // Get the chart data
+        var homeResidentsActivitySumsByType = instance.homeResidentsActivitySumsByType.get();
+        console.log(homeResidentsActivitySumsByType);
```
Comment (brylie) [inline app/client/views/home/activities/sums/sums.js]: Clean up console.log.

File: app/server/methods/activities.js [hunk 5]
```diff
@@ -67,20 +76,60 @@ Meteor.methods({
...
       activityDate: {
-        $gte: previousDate
+        $gte: startDate, $lte: endDate
       }
     };
 
     // Get activties documents cursor
     const activities = Activities.find(query).fetch();
-
+    console.log(activities)
```
Comment (brylie) [inline app/server/methods/activities.js]: Clean up console.log

File: app/server/methods/activities.js [hunk 6]
```diff
@@ -67,20 +76,60 @@ Meteor.methods({
         }
       },
       activityDate: {
-        $gte: previousDate
+        $gte: startDate, $lte: endDate
```
Comment (brylie) [inline app/server/methods/activities.js]: Put each query operator on own line.

File: app/server/methods/homes.js [hunk 7]
```diff
@@ -43,6 +43,20 @@ Meteor.methods({
     // return the resident IDs array
     return residentIds;
   },
+    'getHomeAllResidentIds': function (homeId) {
+        var residents = Residents.find(
```
Comment (brylie) [inline app/server/methods/homes.js]: Use let or const. Default to const, if the reference will not be changing.

File: app/server/methods/homes.js [hunk 8]
```diff
@@ -43,6 +43,20 @@ Meteor.methods({
...
+            {'homeId': homeId},
+            {sort: {firstName: 1}}
+        ).fetch();
+
+        // Create an array containing only resident IDs
+        var residentIds = _.map(residents, function (resident) {
```
Comment (brylie) [inline app/server/methods/homes.js]: Use let or const. Default to const, if the reference will not be changing.

File: app/server/methods/homes.js [hunk 9]
```diff
@@ -78,6 +92,59 @@ Meteor.methods({
 
     return homeActivitiesArray;
   },
+  'getHomeResidentsActivitySumsByType': function (homeId, date = null) {
+    console.log(date);
+    // Get all activity types
+    var activityTypes = ActivityTypes.find({}, {sort: {name: 1}}).fetch();
```
Comment (brylie) [inline app/server/methods/homes.js]: Use let or const. Default to const, if the reference will not be changing.

File: app/server/methods/homes.js [hunk 10]
```diff
@@ -78,6 +92,59 @@ Meteor.methods({
 
     return homeActivitiesArray;
   },
+  'getHomeResidentsActivitySumsByType': function (homeId, date = null) {
+    console.log(date);
```
Comment (brylie) [inline app/server/methods/homes.js]: Clean up console.log

File: app/server/methods/residents.js [hunk 11]
```diff
@@ -70,29 +70,32 @@ Meteor.methods({
     //  sort in reverse order by activity date
     return Activities.find({'residentIds': residentId, activityDate: {$gte: twoWeeksAgo, $lte: now}}, {sort : {activityDate:  -1} });
   },
-  getResidentActivitiesByTypeLast30Days (residentId, activityTypeId) {
-    // Date thirty days ago
-    var thirtyDaysAgo = moment().subtract(30, "days").toDate();
+  'getResidentRecentActivitiesByType': function (residentId, activityTypeId, date) {
+    // Date two weeks ago
+    let startDate = moment().subtract(4, "weeks").toDate();
```
Comment (brylie) [inline app/server/methods/residents.js]: Use `const` if the reference will not change at a later point.

File: app/server/methods/residents.js [hunk 12]
```diff
@@ -70,29 +70,32 @@ Meteor.methods({
...
+    let endDate = new Date();
+    if(date) {
+      startDate = date;
+      endDate = moment(date).endOf('month').toDate();
+    }
     // Get all activities of a specific type involving resident
     // make sure activities are in the past (i.e. not planned)
     //  sort in reverse order by activity date
-    var activities = Activities.find({
+    let activities = Activities.find({
```
Comment (brylie) [inline app/server/methods/residents.js]: Use `const`.

**Comment by brylie** [comment]
Added some minor suggestions about code style. If possible, try to improve code style in the files where you work, even if you didn't write the original code. E.g. replacing `var` with `let`/`const`. Great work so far!

File: app/server/methods/residents.js (hunk 11 above)
Comment (ErikPel) [reply app/server/methods/residents.js]: doesnt the reference change at the line 77?
````
    if(date) {
      startDate = date;
      endDate = moment(date).endOf('month').toDate();
    }

**Comment by ErikPel** [comment]
Should be fine now. It's still saying changes requested but it's because I think the reference changes in that one.

Where do I send the invoice

**Comment by brylie** [comment]
brylie@amble.fi

**Pull request closed without merging**